tasks/post_copy_request_to_queue/test/unit_tests/test_post_copy_request_to_queue.py

```diff
@@ -63,39 +64,51 @@ def tearDown(self):
             "RETRY_BACKOFF": "2",
             "DATABASE_PORT": "5432",
             "DATABASE_NAME": "disaster_recovery",
-            "DATABASE_USER": "rhassan",
+            "DATABASE_USER": "orcauser",
         },
         clear=True,
     )
-    @patch("database.single_query")
-    @patch("requests_db.get_dbconnect_info")
+    @patch("post_copy_request_to_queue.shared_db.get_user_connection")
+    @patch("post_copy_request_to_queue.shared_db.get_configuration")
     @patch("post_copy_request_to_queue.shared_recovery.post_entry_to_queue")
     @patch("post_copy_request_to_queue.shared_recovery.update_status_for_file")
+    @patch("post_copy_request_to_queue.get_metadata_sql")  # todo: Unit test this sql?
     def test_task_happy_path(
-        self,
-        mock_update_status_for_file: MagicMock,
-        mock_post_entry_to_queue: MagicMock,
-        mock_get_db_connect_info: MagicMock,
-        mock_single_query: MagicMock,
+            self,
+            mock_get_metadata_sql: MagicMock,
+            mock_update_status_for_file: MagicMock,
+            mock_post_entry_to_queue: MagicMock,
+            mock_get_db_connect_info: MagicMock,
+            mock_get_user_connection: MagicMock,
     ):
         """
         happy path. Mocks db_connect_info,single_query,
         post_entry_to_queue and update_status_for_file.
         """
-        mock_single_query.return_value = {
-            "job_id": "1",
-            "granule_id": "3",
-            "filename": "f1.doc",
-            "restore_destination": "s3://restore",
-        }
+        # mock_conn = MagicMock()
+        # mock_conn.execute.return_value.fetchall.return_value = [
+        # mock_single_query.begin.return_value.__enter__.return_value = mock_conn
+        # mock_conn.execute = Mock(return_value=[("1", "3", "f1.doc", "restore-bucket")])
+        mock_execute = Mock(return_value=[("1", "3", "f1.doc", "s3://restore")])
+        mock_connection = Mock()
+        mock_connection.execute = mock_execute
+        mock_exit = Mock()
+        mock_enter = Mock()
+        mock_enter.__enter__ = Mock(return_value=mock_connection)
+        mock_enter.__exit__ = mock_exit
+        mock_engine = Mock()
+        mock_engine.begin = Mock(return_value=mock_enter)
+        mock_get_user_connection.return_value = mock_engine
+
         record = self.event["Records"][0]
         new_data = {
             "job_id": "1",
             "granule_id": "3",
             "filename": "f1.doc",
-            "source_key": "b21b84d653bb07b05b1e6b33684dc11b",
-            "target_key": "b21b84d653bb07b05b1e6b33684dc11b",
+            # todo: Value incorrect here and elsewhere. As written, must match mock_execute's return value.
```

Comment:
Looking now, I don't see any other issues where the restore_destination and mock_execute have different values.